### DISCUSSION TITLE
Why include <dht.h> instead of <dht/dht.h>?

### DISCUSSION THREAD

**Discussion by ksh1ng** [UncleRus/esp-idf-lib#404] [Q&A] [answered]
About example of dht, may I know why we include <dht.h> instead of <dht/dht.h>? Is this because of the functionality of esp_idf_lib_helpers used in esp_idf_lib_helpers?

**Comment by UncleRus** [comment, marked as answer]
Please read [Build system section](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/build-system.html#minimal-component-cmakelists) in ESP-IDF documentation. 
In short: the component's CMakeLists.txt specifies the INCLUDE_DIRS parameter equal to the directory of the component itself.

**Reply by ksh1ng** [reply]
Yes, I just saw "INCLUDE_DIRS ." in  idf_component_register(), but I'm curious why <dht/dht.h> causes error in the building process?  Didn't we already set "EXTRA_COMPONENT_DIRS C:/Users/kshsu/Documents/ESP32/IOT_Practice/esp-idf-lib/components"?

**Reply by UncleRus** [reply]
`EXTRA_COMPONENT_DIRS` is not in the list of search pathes. To put it simply, `-I${EXTRA_COMPONENTS_DIRS}/dht` argument passed to the compiler.
